### ISSUE TITLE
Error drawing muscle paths with Arm26 model

### ISSUE THREAD

**Issue by aseth1** [opensim-org/opensim-gui#762]
### Steps to reproduce
- load arm26.osim
- set shoulder elevation to 170degs
- set final time on running dude to 1.0
- simulate

### Expected result
- model simulates and muscle paths are drawn correctly
![image](https://user-images.githubusercontent.com/6494122/40211609-73a594a4-5a00-11e8-8301-79b093ac9651.png)

### Actual result
- TRILong muscle has mysterious muscle path point on/near ground
![image](https://user-images.githubusercontent.com/6494122/40211698-eeee7c20-5a00-11e8-95f2-0ef348dcbef9.png)

### Environment and GUI version
Windows OpenSim e07981e0-2018-05-15

**Comment by aymanhab** [comment]
Debug build runs only to 0.674, muscle is intact. Message in messages window:

ForwardTool::run() caught exception 
SimTK Exception thrown at MassProperties.h:525:
  Error detected by Simbody method Inertia::Inertia(SymMat33): Inertia matrix contains a NaN.
  (Required condition '!isNaN()' was not met.)

The check is enabled only in Debug mode. In release mode it continues quietly causing senseless results downstream.

**Comment by aymanhab** [comment]
@aseth1 How do you want to handle this (exception thrown that stops integration is fired by Simbody in Debug mode only)

**Comment by aymanhab** [comment]
Visualization is a downstream effect of senseless simulation that continued incorrectly. I don't think there's a visualization issue to address here.

**Comment by aymanhab** [comment]
@aseth1 Will report the callstack throwing exception in release (with debug info) then will investigate GUI workflow to reproduce. Either mishandling of exceptions or differences in simbody headers.

**Comment by aseth1** [comment]
@aymanhab there is still a GUI issue. Even in the case that values are NaN, the muscle should not draw. It should not default to some location in ground.

This issues is related to #221 and [core #2141](https://github.com/opensim-org/opensim-core/issues/2141). While agree that the API should not produce NaNs and whatever Tool should terminate immediately, there is no way to guarantee that NaNs won't be generated due to unexpected parameter values and the GUI must handle NaNs appropriately by not drawing the geometry of the related transform.

**Comment by tkuchida** [comment]
> Even in the case that values are NaN, the muscle should not draw. It should not default to some location in ground.

I agree: it's a case of treating `NaN` as a number. The behavior could potentially be very confusing if the muscle path happens to be close to wherever the `NaN` gets drawn.

**Comment by aseth1** [comment]
I verified that same simulation behavior occurs in 3.3. That is to say 3.3 also simulates for the complete 1.0s, but upon replay it only plays for states with numeric values. Loading that model and states file in 4.0 also only played the motion for numeric values.

**Comment by aseth1** [comment]
The error in drawing the path only occurs in the model where I set the default shoulder angle to 170degs. That is the only difference between the two models below. Both are given the same states file and synced. I can play the model individually and get the same behavior (so it rules out syncing).

![image](https://user-images.githubusercontent.com/6494122/40815232-dcbff264-64f9-11e8-846a-e4657be8d5a4.png)

I also use the scripting shell to check that the models compute the same TRILong muscle length at the state where the drawing error occurs. I can only conclude then, that this is purely visualization related.

**Comment by aseth1** [comment]
Further confirming that this is unrelated to forward simulation and the API, is that you can just load the model with the default value set to 170degs and set the shoulder coordinate anywhere between 83 and 131 degs after loading the model and get the TRILong-P2 to draw incorrectly. Again verified that the pathpoint location both in the humerus and in ground are correct.

**Comment by tkuchida** [comment]
Wild! I had to try that myself.
- opened arm26.osim in text editor
- changed `default_value` of "r_shoulder_elev" Coordinate (L311) to 2.967 (approx. 170°) -> [arm26-newDefault.osim.txt](https://github.com/opensim-org/opensim-gui/files/2061701/arm26-newDefault.osim.txt)
- opened model (OpenSim-cedd8d22-2018-05-31-win64.exe)
- in *Coordinates* panel, changed "r_shoulder_elev" to 83

![](https://user-images.githubusercontent.com/4203505/40829133-731db34e-6537-11e8-9791-9ca3900b0041.png)

**Comment by aymanhab** [comment]
This was caused by wrapping returning 1 point which was treated as error condition in the past. Made more lenient now.

**Comment by aseth1** [comment]
>  This was caused by wrapping returning 1 point which was treated as error condition in the past. Made more lenient now.

Wrapping should not return 1 point unless it is perfectly tangent to the wrap object.  If it was treated as an error condition in the past, how does making it more lenient supposed to fix this problem?

**Comment by aseth1** [comment]
Confirmed on Windows OpenSim_cea85df3-2018-07-18 that setting the default value to 170degs and reposing to 83degs does not have the above issue.

However, the original issue (following the steps at the top of this issue) and simulating does cause similar problems in drawing muscles at 0.760s at which time the states become NaN:

![image](https://user-images.githubusercontent.com/6494122/43295918-073cb91c-90fd-11e8-8d7a-d406d81633a7.png)

You can see that the states have no values beyond 0.760 but the muscle paths are still attempting to be drawn. In this case the distal (undefined) points are being set to (0,0,0) in ground which is incorrect behavior.

**Comment by aymanhab** [comment]
@aseth1 thanks for checking. The key question is how to stop the simulation when the state contains NaNs.  The only mechanism the GUI knows to stop accumulating data and using NaNs is an Exception being thrown. This happens in debug mode by Simbody and not in release for efficiency purposes. Does the state have a way to quick check if it contains NaNs?

**Comment by tkuchida** [comment]
> how to stop the simulation when the state contains NaNs

Not sure the simulation should stop if NaNs are encountered. Is there a way to prevent the visualizer from changing NaNs to zeros? A user could potentially load results from the API or an old version of OpenSim, so it seems like there needs to be a strategy for handling NaNs in the visualizer regardless.

**Comment by aseth1** [comment]
There is an inconsistency here as well. The bones correctly do not display because the transforms are NaN. The fact that muscle paths still attempt to draw is a bug. In 3.3 the arm (downstream of NaN coordinates) disappears, which I believe is the correct behavior.
